libpod/stats_linux.go

```diff
@@ -36,6 +37,11 @@ func (c *Container) getPlatformContainerStats(stats *define.ContainerStats, prev
 	// Ubuntu does not have swap memory in cgroups because swap is often not enabled.
 	cgroupStats, err := cgroup.Stat()
 	if err != nil {
+		// cgroup.Stat() is not an atomic operation, so it is possible that the cgroup is removed
+		// while Stat() is running.  Try to catch this case and return a more specific error.
+		if (errors.Is(err, unix.ENOENT) || errors.Is(err, unix.ENODEV)) && !cgroupExist(cgroupPath) {
+			return fmt.Errorf("cgroup %s does not exist: %w", cgroupPath, define.ErrCtrRemoved)
```

Comment:
define.ErrCtrRemoved doesn't sound like the proper error to return, that means the container has been removed but that does not have to be true. Maybe it was just stopped. So I think it should return define.ErrCtrStopped which is also already skipped
```
						// All these errors might happen while we get stats, when we list all
						// they must be skipped as they cause podman stats to stop and error otherwise.
						// ErrCtrStopped can happen when the container process exited before we could
						// update the container state
						// https://github.com/containers/podman/issues/23334
						(errors.Is(err, define.ErrCtrRemoved) || errors.Is(err, define.ErrNoSuchCtr) ||
							errors.Is(err, define.ErrCtrStateInvalid) || errors.Is(err, define.ErrCtrStopped)) {
						continue
					}
```

Reply:
good observation.  I've fixed it and pushed a new version